dot/services_test.go

```diff
@@ -0,0 +1,639 @@
+// Copyright 2021 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package dot
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/ChainSafe/gossamer/dot/core"
+	"github.com/ChainSafe/gossamer/dot/digest"
+	"github.com/ChainSafe/gossamer/dot/network"
+	"github.com/ChainSafe/gossamer/dot/rpc"
+	"github.com/ChainSafe/gossamer/dot/state"
+	"github.com/ChainSafe/gossamer/dot/sync"
+	"github.com/ChainSafe/gossamer/dot/system"
+	"github.com/ChainSafe/gossamer/dot/telemetry"
+	"github.com/ChainSafe/gossamer/dot/types"
+	"github.com/ChainSafe/gossamer/internal/log"
+	"github.com/ChainSafe/gossamer/lib/babe"
+	"github.com/ChainSafe/gossamer/lib/genesis"
+	"github.com/ChainSafe/gossamer/lib/grandpa"
+	"github.com/ChainSafe/gossamer/lib/keystore"
+	"github.com/ChainSafe/gossamer/lib/runtime"
+	"github.com/ChainSafe/gossamer/lib/runtime/life"
+	rtstorage "github.com/ChainSafe/gossamer/lib/runtime/storage"
+	"github.com/ChainSafe/gossamer/lib/runtime/wasmer"
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func Test_createBlockVerifier(t *testing.T) {
+	cfg := NewTestConfig(t)
+
+	genFile := newTestGenesisRawFile(t, cfg)
+
+	cfg.Init.Genesis = genFile
+
+	nodeInstance := nodeBuilder{}
+	err := nodeInstance.initNode(cfg)
+	require.NoError(t, err)
+
+	stateSrvc, err := nodeInstance.createStateService(cfg)
+	require.NoError(t, err)
+
+	stateSrvc.Block = &state.BlockState{}
+	stateSrvc.Epoch = &state.EpochState{}
+
+	type args struct {
+		st *state.Service
+	}
+	tests := []struct {
+		name string
+		args args
+		want *babe.VerificationManager
```

Comment:
have this as `expectNil bool` if we don't assert it against the returned babe verification manager.

Same for below tests

Reply:
Updated.